aristanetworks/ansible-avd
# Feat(eos_config_deploy_cvp_v2): New role for deployment to CVP

## Change Summary

create v2 role for eos_config_deploy_cvp that uses `delegate_to` knob in ansible to run tasks in parallel

## Related Issue(s)

Fixes #1544 

## Component(s) name

`arista.avd.eos_config_deploy_cvp_v2`

## Proposed changes



## How to test

example playbook:

```
---
- name: Build Switch configuration
  hosts: TP_FABRIC
  connection: local
  gather_facts: no

  tasks:
    - name: generate intented variables
      tags: [build]
      import_role:
         name: arista.avd.eos_designs

    - name: generate device intended config and documention
      tags: [build]
      import_role:
         name: arista.avd.eos_cli_config_gen

- name: Configuration deployment with CVP
  hosts: TP_FABRIC
  connection: local
  gather_facts: no
  tasks:
    - name: run CVP provisioning
      import_role:
         name: arista.avd.eos_config_deploy_cvp_v2
      vars:
        container_root: 'TP_FABRIC'
        configlets_prefix: 'tp-avd'
        #cv_collection: v3
        device_filter: 'tp-avd'
        state: present
```
note that `hosts` has to be set to the fabric instead of the CVP server, so for each device we'd create a separate session against CVP

## Checklist

### User Checklist


- N/A

### Repository Checklist



- [ x ] My code has been rebased from devel before I start
- [ x ] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ x ] My change requires a change to the documentation and documentation have been updated accordingly.
